### ISSUE TITLE
BUG STILL HAPENNING

### ISSUE THREAD

**Issue by batata004** [kalcaddle/KodExplorer#149]
Hi my friend,

The bug with characters that I already reported 4 times is still happening. Every single time I report it you say you corrected it, so I install everything and realize the problem is still going. Upload the attached file to your server and use the latest version of koeplorer. Try to edit.
[index.txt](https://github.com/kalcaddle/KodExplorer/files/970601/index.txt)

When I try to edit it it shows:

![image](https://cloud.githubusercontent.com/assets/8988923/25619876/d10fdd98-2f22-11e7-9f2e-b1fe36d66e34.png)

But it should show some special characters. ANY OTHER ONLINE EDITOR I use to edit files (like the one I attached you) work perfectly, wheter I use tinymce or any other major online editor. I have no idea why your code keeps crasching and having problems.

**Comment by kalcaddle** [comment]
File encoding problem,
it is recommended to use the unified UTF-8 coding
PHP automatically identify the coding function mb_detect_encoding，sometimes inaccurate. The new version of the manual selection of the file code, manually select the file code


![image](https://cloud.githubusercontent.com/assets/3761968/25645690/53325a9c-2fe3-11e7-84cf-259a18237b9e.png)

**Comment by kalcaddle** [comment]
Each country's various file code, is a headache problem.No a unified solution, we have to try to do the maximum automatic processing.

**Comment by batata004** [comment]
I dont have this kind of problem with any any other editor.

ALSO with the version of early 2016 I never had this problm. NEVER. This problem started happening about 6 to 8 months ago What is going on? Ok, if you dont wanna fix this, sooner or later other people will complain so I will move to FILERUN which is way better and has really few bugs.

**Comment by kalcaddle** [comment]
Please understand that there is only my personal development and maintenance of this project. I have been trying to make the program better and more stable.
File code issues The next version will add automatic identification and user manual selection, so as to avoid garbled as much as possible.

I tried filerun and found something similar to our previous version. [your file:index.txt]
![image](https://cloud.githubusercontent.com/assets/3761968/25731938/50e4d9ae-317d-11e7-9169-c5f6b44507ca.png)

**Comment by batata004** [comment]
My freiend, I dont want to be disrespectful I really appreciate your attention into this. And you are doing all by yourself and for free, I cant complain or offend you. It's not fair, you are being so kind providing such a powerful tool and really good BTW.

But this problem with the file encoding is driving me crazy. Could you please install in a server some old version (from the beginning of last year from JANUARY for example) and open the file I provided you? I ask you to install the bare version of your code from JANUARY/2016 then go to the file 

config/config.php

And change this

$config['check_charset'] = 'ASCII,UTF-8,GBK...

To this:

$config['check_charset'] = 'ISO-8859-1,ASCII,UTF-8,GBK...

Upload this new config file to your server. After that you can easily edit exacly this file I provided you. 

You see, this problem didnt happen in old version with the above change.

Regarding FILERUN I am using an old version of them in another server and it works really fine, I didnt have the chance to test the new version of their code to see if now they are screwing up the file content. But in the old version I was using it worked perfectl.

Could you please do the test above? Unfortunatelly I dont have the old KODEXPLORER.ZIP file only the new that I downoad from GITHUB, so could you find a backup in your system of the old version and try it out, please?

**Comment by batata004** [comment]
could you replicate the test my friend?

**Comment by dagnelies** [comment]
Like @kalcaddle said, this is a "universal" problem regarding what encoding is used. Basically, how you encode text as bytes and vice-versa. Now, the tricky part is, that it is *impossible* to know which encoding was used by looking at the bytes. Some editors try to "guess" what encoding is *likely*, but it's all more or less flaky.

Instead of:
`$config['check_charset'] = 'ISO-8859-1,ASCII,UTF-8,GBK...`
I would also recommend to use:
`$config['check_charset'] = 'ASCII,UTF-8,ISO-8859-1,GBK...`
because it's much more likely to misinterpret bytes as `ISO-8859-1` rather than `UTF-8`.

Lastly, the different behavior since 2016 *might* not be related to the code, but related to libraries or the system environment. I'm already amazed that kodexplorer supports recognition of multiple encodings.

**Comment by batata004** [comment]
@dagnelies I didnt mean to offend the developer of this great library, he is doing a really amazing job, really good and the best part of his plataforma is that it does not require any database, that's amazing.

Regarding the encoding, in the version of the last year I didnt have any problem, so in my opinion the algorightm used was pretty better. It may be a problem with a component/plugin that was changed in the newer code (and it's not fault of the dev of kodexplorer) but it worth a investigation cause certainly, with no questions, the version of 2016 was perfect, I can say it : PERFECT! I never had a problem with the encoding. From there to the newer versions I only had problems, I already opened more than 6 threads related to this in GIT.

**Comment by kalcaddle** [comment]
We will fixed this problem at next version.
Please have a wait.

**Comment by batata004** [comment]
Thank you so much man! Thank you!!! Anyway, just an advice, you should take a look at the versio of the beginning of 2016 cause in that version I never had problems with characeters if I used the line below:

$config['check_charset'] = 'ISO-8859-1,ASCII,UTF-8,GBK

**Comment by dagnelies** [comment]
@batata004 Btw, to help him, you could at least try out the various versions from beginning 2016 so that you can pin point which exact version caused the change ...and also to be sure it does not depend on something else on your system.

**Comment by batata004** [comment]
As I said earlier I dont have access to old versions. Do you?

**Comment by dagnelies** [comment]
https://github.com/kalcaddle/KodExplorer/commits/master

**Comment by batata004** [comment]
Great! I didnt know it! Really great! I installed the 3.21 version and unfortunatelly I got this error:

Uncaught SyntaxError: Invalid regular expression: missing /
Uncaught ReferenceError: seajs is not defined(anonymous function) 

I think I saw this error in the past but have no idea how I fixed it. Anyway it's gonna but hard to find what is causing this cause I remember it took me some days to fix this.
